src/examples/TDXContainers.sol

```diff
@@ -0,0 +1,189 @@
+        owner = msg.sender;
+        keymgr = _keymgr;
+
+        /* TODO: should we allow more than one pck? */
+        pckCertPem = _pckCertPem;
+        pckCrlPem = _pckCrlPem;
+
+        /* TODO: allow more than one tcbInfo */
+        tcbInfo = _tcbInfo;
+
+        qeIdentity = _qeIdentity;
+
+        /* TODO (maybe): configure ca chain (root ca & crl, pck chain) */
+    }
+
+    function onchain_onboardHost(bytes32 _quoteMeasurementHash, bytes memory _hostPubkey, bytes memory att) external {
+        require(msg.sender == owner);
+
+        require(quoteMeasurementHash == bytes32(0)); // Only once
+        require(keymgr.verify(address(this), keccak256(abi.encodePacked("hPub", _hostPubkey, "qM", _quoteMeasurementHash)), att));
+
+        quoteMeasurementHash = _quoteMeasurementHash;
+        hostPubkey = _hostPubkey;
+
+        /* TODO: mark start/restart of the host */
+        /* TODO: mark all containers stopped */
+    }
+
+    /* Hash of the pubkey passed in is expected to be in rtmr3 */
+    function offchain_onboardHost(bytes memory quote, bytes memory _hostPubkey) external returns (bytes32, bytes memory, bytes memory) {
+        require(msg.sender == owner);
+
+        require(quoteMeasurementHash == bytes32(0)); // Only once
+
+        bytes1 quote_version = quote[0];
+        bytes memory quoteMeasurement = new bytes(144);
+        bytes1[48] memory rtmr3;
+        if (quote_version == 0x04) {
+            for (uint i = 184; i < 184+48; i++) { /* mrTd */
+                quoteMeasurement[i-184] = quote[i];
+            }
+            for (uint i = 376; i < 376+144; i++) { /* rtmr 0-2 */
+                quoteMeasurement[i+32-376] = quote[i];
+            }
+            for (uint i = 376+144; i < 376+144+48; i++) {
+                rtmr3[i-376-144] = quote[i];
+            }
+        } else if (quote_version == 0x05) {
+            /* TODO: check body type (doesnt matter for now) */
+            for (uint i = 184+6; i < 184+6+48; i++) { /* mrTd */
+                quoteMeasurement[i-184-6] = quote[i];
+            }
+            for (uint i = 376+6; i < 376+6+144; i++) { /* rtmr 0-2 */
+                quoteMeasurement[i+32-376-6] = quote[i];
+            }
+            for (uint i = 376+6+144; i < 376+6+144+48; i++) {
+                rtmr3[i-376-144-6] = quote[i];
+            }
+        } else revert("unsupported quote version");
+
+        bytes32 _quoteMeasurementHash = keccak256(quoteMeasurement);
+        /* TODO: verify measurements are as expected! */
+        /* TODO: require(_quoteMeasurementHash == _expectedMeasurement); */
+
+        bytes32 hostPubkeyHash;
+        assembly {
+            hostPubkeyHash := mload(add(rtmr3, 32))
+        }
+        require(hostPubkeyHash == keccak256(_hostPubkey));
+
+        keymgr.Suave().verifyTDXDCAPQuote(quote, pckCertPem, pckCrlPem, tcbInfo, qeIdentity);
```

Comment:
perfect - the TDX attestation in this application avoids having to be online! Although this example uses KeyManager for bootstrapping-the-bootstrap, we could probably switch the whole KeyManager to use this pattern